File: peer-fixtures/opentui/sb7_resize_storm/bin/sb7_resize_storm_benchmark.js
#!/usr/bin/env bun

import { createCliRenderer, TextRenderable } from '@opentui/core'

const options = parseArgs(process.argv.slice(2))
await runWire(options)

async function runWire(options) {
  const renderer = await createCliRenderer({
    width: options.terminalColumns,
    height: options.terminalRows,
    consoleMode: 'disabled',
    screenMode: 'alternate-screen',
    clearOnShutdown: true,
    targetFps: 60,
    maxFps: 60,
    exitOnCtrlC: false,
  })
  const state = { step: 0 }
  const text = new TextRenderable(renderer, {
    id: 'sb7-resize-storm',
    width: options.terminalColumns,
    height: options.terminalRows,
    content: renderBody(options, state),
    wrapMode: 'none',
    truncate: true,
  })
  renderer.root.add(text)
  try {
    for (let step = 0; step <= options.steps; step += 1) {
      state.step = step
      text.content = renderBody(options, state)
      renderer.requestRender()
      await renderer.idle()
      await sleep(options.intervalMs)
    }
  } finally {
    renderer.destroy?.()
  }
}

function renderBody(options, state) {
  const width = process.stdout.columns || options.terminalColumns
  const height = process.stdout.rows || options.terminalRows
  const visibleLogs = Math.max(2, Math.min(10, Math.floor(height / 3)))
  const visibleRows = Math.max(3, Math.min(14, height - visibleLogs - 4))
  const lines = [
    `SB.7 resize step=${state.step} rows=${options.rows} size=${width}x${height}`,
    'filter status:failed',
    '',
  ]
  for (let row = 0; row < visibleRows; row += 1) {
    const index = (state.step * 7 + row) % options.rows
    lines.push(
      `RUN-${100000 + index} ${status(index).padEnd(8)} owner=${owner(index).padEnd(5)} duration=${String(index % 3).padStart(2, '0')}:${String(index % 60).padStart(2, '0')} Resize shard ${index % 2048}`,
    )
  }
  lines.push('')
  for (let row = 0; row < visibleLogs; row += 1) {
    const index = state.step * visibleLogs + row
    const unsafe = index % 17 === 0 ? ` secret-${index} payload` : ''
    lines.push(`resize log ${index} shard=${index % 31} status=${status(index)}${unsafe}`)
  }
  return lines.join('\n')
}

function status(row) {
  return ['queued', 'running', 'passed', 'failed', 'blocked'][row % 5]
}

function owner(row) {
  return ['agent', 'ops', 'qa', 'infra', 'cli'][row % 5]
}

function parseArgs(args) {
  const options = {
    rows: 100000,
    steps: 8,
    intervalMs: 80,
    terminalColumns: 120,
    terminalRows: 32,
  }
  for (const arg of args) {
    if (arg === '--wire') continue
    if (arg.startsWith('--rows=')) options.rows = positive(arg, '--rows=')
    else if (arg.startsWith('--steps=')) options.steps = positive(arg, '--steps=')
    else if (arg.startsWith('--interval-ms=')) options.intervalMs = positive(arg, '--interval-ms=')
    else if (arg.startsWith('--size=')) {
      const [columns, rows] = arg.slice('--size='.length).split('x').map(Number)
      if (!Number.isInteger(columns) || columns <= 0 || !Number.isInteger(rows) || rows <= 0) {
        throw new Error('--size must be COLSxROWS')
      }
      options.terminalColumns = columns
      options.terminalRows = rows
    } else {
      throw new Error(`unknown argument: ${arg}`)
    }
  }
  return options
}

function positive(arg, prefix) {
  const value = Number(arg.slice(prefix.length))
  if (!Number.isInteger(value) || value <= 0) throw new Error(`${prefix} expects a positive integer`)
  return value
}

function sleep(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}

File: peer-fixtures/textual/sb7_resize_storm/sb7_resize_storm_benchmark.py
from __future__ import annotations

import argparse
import asyncio

from textual.app import App, ComposeResult
from textual.widgets import Static


def main() -> None:
    options = parse_args()
    if options.wire:
        WireResizeStormApp(options).run()
    else:
        print("Textual SB.7 resize storm fixture: use --wire for PTY capture")


class WireResizeStormApp(App[None]):
    def __init__(self, options: argparse.Namespace) -> None:
        super().__init__()
        self.options = options
        self.step = 0

    def compose(self) -> ComposeResult:
        yield Static(self.render_body(), id="body")

    def on_mount(self) -> None:
        self.body = self.query_one("#body", Static)
        asyncio.create_task(self.drive())

    async def drive(self) -> None:
        await asyncio.sleep(self.options.interval_ms / 1000)
        while self.step < self.options.steps:
            self.step += 1
            self.body.update(self.render_body())
            await asyncio.sleep(self.options.interval_ms / 1000)
        self.exit()

    def render_body(self) -> str:
        width, height = self.size
        visible_logs = max(2, min(10, height // 3))
        visible_rows = max(3, min(14, height - visible_logs - 4))
        lines = [
            f"SB.7 resize step={self.step} rows={self.options.rows} size={width}x{height}",
            "filter status:failed",
            "",
        ]
        for row in range(visible_rows):
            index = (self.step * 7 + row) % self.options.rows
            lines.append(
                f"RUN-{100000 + index} {status(index):8} owner={owner(index):5} "
                f"duration={index % 3:02d}:{index % 60:02d} Resize shard {index % 2048}"
            )
        lines.append("")
        for row in range(visible_logs):
            index = self.step * visible_logs + row
            unsafe = f" secret-{index} payload" if index % 17 == 0 else ""
            lines.append(
                f"resize log {index} shard={index % 31} status={status(index)}{unsafe}"
            )
        return "\n".join(lines)


def status(row: int) -> str:
    return ["queued", "running", "passed", "failed", "blocked"][row % 5]


def owner(row: int) -> str:
    return ["agent", "ops", "qa", "infra", "cli"][row % 5]


def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser()
    parser.add_argument("--wire", action="store_true")
    parser.add_argument("--rows", type=positive_int, default=100_000)
    parser.add_argument("--steps", type=positive_int, default=8)
    parser.add_argument("--interval-ms", type=positive_int, default=80)
    parser.add_argument("--size", default="120x32")
    return parser.parse_args()


def positive_int(value: str) -> int:
    parsed = int(value)
    if parsed <= 0:
        raise argparse.ArgumentTypeError("expected a positive integer")
    return parsed


if __name__ == "__main__":
    main()
